content/use-cases/contribute-code-to-ripple-lib.md

```diff
@@ -0,0 +1,69 @@
+## <a href="https://github.com/ripple/ripple-lib" target="_blank">Access the `ripple-lib` repo <i class="fa fa-external-link" aria-hidden="true"></i></a> <!--#{ fix for md highlighting_ #}-->
+
+`ripple-lib` is an open-source project. You can take a look at `ripple-lib` code simply by accessing the `ripple-lib` GitHub repo. Before contributing or reporting bugs, we recommend that you get to know the code and developer experience by performing the following tasks.
+
+
+<span class="use-case-step-num">{{n.next()}}</span>
+<!-- <span class="use-case-step-length">(1 hour)</span> -->
+## [Set up and run a `rippled` server](manage-the-rippled-server.html)
+
+RippleAPI is an API for interacting with the XRP Ledger. The core peer-to-peer server that manages the XRP Ledger is `rippled`. Set up and run a `rippled` server to understand its developer experience and functionality. Anyone can run their own `rippled` server that follows the network and keeps a complete copy of the XRP Ledger.
+
+
+<span class="use-case-step-num">{{n.next()}}</span>
+<!-- <span class="use-case-step-length">(1 hour)</span> -->
+## [Get a Test Net XRP Ledger account](xrp-test-net-faucet.html)
+
+Use the XRP Ledger Test Net to get a test account. Connect your `rippled` server to the Test Net to make test calls and get to know the XRP Ledger.
+
+
+<span class="use-case-step-num">{{n.next()}}</span>
+<!-- <span class="use-case-step-length">(1 hour)</span> -->
+## [Set up your `ripple-lib` development environment](get-started-with-rippleapi-for-javascript.html#environment-setup)
+
+`ripple-lib` requires Node.js and a few dependencies that you can get using npm (Node Package Manager). Also, be sure to create your own fork of the `ripple-lib` repository on GitHub so you can contribute pull requests to the official repo.
+
+
+<span class="use-case-step-num">{{n.next()}}</span>
+<!-- <span class="use-case-step-length">(1 hour)</span> -->
+## [Run your first `ripple-lib` script](get-started-with-rippleapi-for-javascript.html#first-rippleapi-script)
+
+Examine and run the `get-account-info.js` script. Use it to get a feel for how RippleAPI scripts work and to verify that your RippleAPI interface is working.
+
+
+<span class="use-case-step-num">{{n.next()}}</span>
+<!-- <span class="use-case-step-length">(1 hour)</span> -->
+## Familiarize yourself with `ripple-lib`'s coding style<!--#{ link up once we have some a ripple-lib coding style page #}-->
+
+Before you start contributing code to `ripple-lib,` take some time to familiarize yourself with the coding standards used in the `ripple-lib` repo. These standards gradually evolve and propagate through code reviews. Some aspects are enforced more strictly than others. ***TODO: Here are the coding standards for rippled - specific to C++: https://github.com/ripple/rippled/blob/develop/docs/CodingStyle.md. Do we want to modify for JavaScript and ripple-lib and provide it in the ripple-lib repo?***
```

Comment:
@intelliot - do we want to provide ripple-lib-specific coding style info before we publish this use case? Or should we publish the use case without this step -- and update later? Any thoughts?

Reply:
We have a linter (`tslint`) that catches most inconsistencies automatically. The rest get caught in code review. I'm fine with leaving out mention of a coding style for now; it hasn't been a problem yet.